pkg/minikube/cluster/cluster.go

```diff
@@ -248,12 +255,33 @@ func (m *MachineConfig) CacheMinikubeISOFromURL() error {
 		}()
 	}
 
+	if _, err = io.Copy(tmpISOFile, iso); err != nil {
+		return err
+	}
+
+	if err := tmpISOFile.Sync(); err != nil {
+		return err
+	}
+
+	checksumURL := fmt.Sprintf(m.MinikubeISO + ".sha256")
+	checksumResp, err := http.Get(checksumURL)
+	defer checksumResp.Body.Close()
+	if err != nil {
+		return err
+	}
+	if checksumResp.StatusCode != 404 {
```

Comment:
I think that might fall a bit short. There might be many other things which can go wrong. 404 means file not found, but there might be other HTTP errors. I think you need to define this the other around, anything except a 200 is in this context a failure.